src/main/java/com/artipie/npm/proxy/NpmProxy.java

```diff
@@ -83,130 +59,64 @@
      * @param storage Adapter storage
      */
     public NpmProxy(final NpmProxyConfig config, final Vertx vertx, final Storage storage) {
-        this.config = config;
+        this(
+            vertx,
+            new RxNpmProxyStorage(new RxStorageWrapper(storage)),
+            new HttpNpmRemote(config, vertx)
+        );
+    }
+
+    /**
+     * Default-scoped ctor (for tests).
+     * @param vertx Vertx instance
+     * @param storage NPM storage
+     * @param remote Remote repository client
+     * @checkstyle ParameterNumberCheck (10 lines)
+     */
+    NpmProxy(final Vertx vertx,
```

Comment:
@brastak we're trying to avoid explicit `Vertx` dependency in other adapters and prefer abstractions, like `Storage` which hides `Vertx` usage inside and provides only interface methods. In long term plans we may create custom HTTP server implementation and get rid of `Vertx` dependency, with explicit Vertx dependency it would be hard to replace it.

Reply:
@g4s8 I use Vertx here to get Vertx Web Client. It is not storage dependency, it is dependency of adapter itself